dist/stat_data.py

```diff
@@ -808,6 +809,8 @@ def __init__(self, name, desc, flags=''):
     BlockStat('block_minor', 'minor version number', 'max_aggregate,no_scale'),
     BlockStat('block_reuse_bytes', 'file bytes available for reuse', 'no_scale,size'),
     BlockStat('block_size', 'file size in bytes', 'no_scale,size'),
+    BlockStat('block_total_padding_size', 'block total padding size', 'no_scale,size'),
+    BlockStat('block_total_write_size', 'block total write size', 'no_scale,size'),
```

Comment:
I don't think these stats are useful: as they do not account for any preexisting padding in the file, even if the file is newly created in the session, if compaction is enabled it may well invalidate these stats.
This feels like something better suited to `wt` if not supported.

Reply:
Even if compression is enabled, these are the real disk footprint after compression. Here I personally think the session dimension statistics will be more useful than the conn dimension statistics, because we can see the overall padding percentage of each table.
For example, if we have a mongod node with 100 tables, how do we determine which table the padding is caused by?

We find that the disk padding of the entire mongod node is high, and how can we quickly determine which table is causing the padding? Therefore, I think the statistics of session dimension (that is, table dimension statistics) is very useful in combination with the real usage scenarios of users.

Reply:
Another benefit of the block_total_write_size configuration is that if at some point in time the mongod node is jitter due to a disk IO bottleneck, we can determine which table is the main cause and which table is using the most disk IO resources.

Reply:
Can these statistics not be derived by running a tool the files containing the tables? Is it necessary to collect and log the statistics dynamically?

Reply:
1. As far as I know, wt tool needs to shut down mongod node to do analysis. (I'm not sure if wt supports not closing the mongod node for analysis here, From the logic of the code, we need to shut down mongod node to analyze, because the wt directory can only be used by one process at a time.)
2. The current wt tool should not be able to analyze, and it needs to improve the wt code to support it
3. Even if the wt tool supports analysis, it is not recommended to use it in online environments. This will affect users' access to the mongod node, because it will increase the system I/O load. I think the wt tool is an offline analysis tool rather than an online analysis tool.

For these reasons, I think this statistic is useful.
Thanks.
:)

Reply:
I'm not familiar enough with the `wt` tool / system to answer with confidence, but I believe it does need to be offline and I can see the inherent disadvantage.

So what i hope is my last question is: can the block and padding data be collected and totaled by a tool outside of WT? 

This would allow the mongodb node to remain offline, but offload the work. I appreciate this seems like a trivial piece of work, but these trivial pieces of work accumulate, if we can keep the work to a useful minimum we have a better chance of maintaining and improving performance.

Thanks.

Reply:
So what i hope is my last question is: can the block and padding data be collected and totaled by a tool outside of WT?

In theory, perfecting wt tool is available, because the end padding of a block ext is all zeros and can theoretically be recognized.

Reply:
Hi yyz, sorry I'll need more time to consult with a colleague who is more familiar with block storage.

In the interim: it was pointed out to me that if the goal is to tune the block size, this would require converting the entire database to use that new block size. So I am struggling to understand why this data is required for a running system.

Thanks.

Reply:
No problem. Thanks for your reply.
This does not represent the current real-time statistics for block, but the historical operation statistics related to block. This is similar to the existing block_byte_write statistics, for example, block_byte_write represents the total number of blocks written in the past. 
This is how I understand it.
:)